data_sync/mvds.md

```diff
@@ -0,0 +1,133 @@
+# Minimum Viable Data Synchronization
+
+**DRAFT VERSION 0.5**
+
+*Written by Oskar Thorén oskar@status.im & Dean Eigenmann dean@status.im*
+
+## Table of Contents
+
+1. [Abstract](#abstract)
+2. [Definitions](#definitions)
+3. [Wire Protocol](#wire-protocol)
+    1. [Secure Transport](#secure-transport) 
+    2. [Payloads](#payloads)
+4. [Synchronization](#synchronization)
+    1. [State](#state)
+    2. [Flow](#flow)
+    3. [Retransmission](#retransmission)
+5. [Footnotes](#footnotes)
+6. [Acknowledgements](#acknowledgements)
+
+## Abstract
+
+In this specification, we describe a minimum viable protocol for data synchronization inspired by the Bramble Synchronization Protocol<sup>1</sup>. This protocol is designed to ensure reliable messaging between peers across an unreliable peer-to-peer (P2P) network where they may be unreachable or unresponsive.
+
+We present a functional specification for future implementation<sup>2</sup> as well as reference simulation data which demonstrates its performance.
+
+## Definitions
+
+| Term       | Description                                                                         |
+|------------|-------------------------------------------------------------------------------------|
+| **Peer**   | The other nodes that a node is connected to.                                        |
+| **Record** | Defines a payload element of either the type `OFFER`, `REQUEST`, `MESSAGE` or `ACK` |
+| **Node**   | Some process that is able to store data, do processing and communicate for MVDS.    |
+
+## Wire Protocol
+
+### Secure Transport
+
+This specification does not define anything related to the transport of packets. It is assumed that this is abstracted in such a way that any secure transport protocol could be easily implemented. Likewise, properties such as confidentiality, integrity, authenticity and forward secrecy are assumed to be provided by a layer below.
+
+### Payloads
+
+Payloads are implemented using [protocol buffers](https://developers.google.com/protocol-buffers/).
+
+```protobuf
+message Payload {
+  repeated bytes acks = 1;
+  repeated bytes offers = 2;
+  repeated bytes requests = 3;
+  repeated Message messages = 4;
+}
+
+message Message {
+  bytes group_id = 1;
+  int64 timestamp = 2;
+  bytes body = 3;
+}
+
+```
+
+Each payload contains the following fields:
+
+- **Acks:** This field contains a list (can be empty) of `message identifiers` informing the recipient that sender holds a specific message.
```

Comment:
what's a `message identifier`?

Reply:
Will add documentation its the `HASH(group_id, timestamp, body)` where `HASH` is some hash function.